src/components/RightHandMenu/DonutChart/Donut.js

```diff
@@ -0,0 +1,153 @@
+
+  const myCalc = (midAngle, sin, sy) => {
+    if (
+      midAngle <= 10 ||
+      midAngle >= 350 ||
+      (midAngle >= 170 && midAngle <= 190)
+    ) {
+      return sy
+    } else if (midAngle >= 340 || (midAngle >= 180 && midAngle <= 220)) {
+      return cy + (outerRadius + 30) * sin + 20
+    } else if (midAngle <= 20 || (midAngle >= 140 && midAngle <= 180)) {
+      return cy + (outerRadius + 30) * sin - 20
+    } else {
+      return cy + (outerRadius + 30) * sin
+    }
+  }
+
+  const sin = Math.sin(-RADIAN * midAngle)  //convert degree to radian units and find sin value of it
+  const cos = Math.cos(-RADIAN * midAngle)  //convert degree to radian units and find cos value of it
+  const sx = cx + outerRadius * cos         //start coordinates (sx,sy) for line
+  const sy = cy + outerRadius * sin
+  const mx = midAngle <= 10 || midAngle >= 350 || (midAngle >= 170 && midAngle <= 190) //mid coordinates (mx,my) for line
+              ? cx + (outerRadius + 30) * cos
+              : sx
+  const my = myCalc(midAngle, sin, sy)
+  const ex = midAngle <= 10 || midAngle >= 350 || (midAngle >= 170 && midAngle <= 190)  //end coordinates (ex,ey) for line
+              ? mx
+              : mx + (cos >= 0 ? 1 : -1) * 22   //1st and 4th quadrant cos is +ve ; 1st and 2nd quadrant sin is +ve
+  const ey = my
+  const textAnchor = cos >= 0 ? 'end' : 'start' //where to anchor text tag
+  const tx= ex + (cos >= 0 ? 1 : -1) * 12       //coordinates (tx,ty) on which text tag is placed
+  const ty= ey + (sin >= 0 ? 1 : -1) * 16
+
+
+  return (
+    <g>
+      {/* centers text in middle of pie chart */}
+      {/* <text x={cx} y={cy} dy={8} textAnchor="middle" fill={fill}>
+        {payload.name}
+        </text> */}
+
+      {/* line from pie chart to label name */}
+      <path
+        d={`M${sx},${sy}L${mx},${my}L${ex},${ey}`}
+        stroke={fill}
+        fill='none'
+      />
+
+      {/* Circle at the tip of line */}
+      <circle cx={ex} cy={ey} r={3} fill={fill} stroke='none' />
+
+      {/* Label name/ text at end tip of line from pie chart */}
+      <text
+        textLength='30'
+        x={tx}
+        y={ty}
+        textAnchor={textAnchor}
+      >
+        <tspan 
+          fill='#1c752a'
+          x={tx}
+          y={ty} >{`${data[name].key}`}</tspan>
+        <tspan
+          className='tspan__val'
+          x={tx + (cos >= 0 ? -1 : 1) * 6}
+          y={ty + 10}>{`${value}`}</tspan>
+      </text>
+    </g>
+  )
+}
+
+
+/*
+ * COMPONENT: Donut
+ * for Race and similar type data
+ * data slice filtered as per radioSelct and ToggleSelect to be used in place of mockData
+ */
+function Donut() {
+  const [legend, setLegend] = useState([])
+  const [sum, setSum] = useState(0)
+
+  useEffect(() => {
+    const l = []
+    let sum1 = data.reduce(function (a, b) {
+      return a + b.value
+    }, 0)
+    data.map((entry, index) => (l.push({ 'key' : entry.key, 'color' : COLORS[index % COLORS.length], 'value' : entry.value  })))
```

Comment:
Is there a reason you are using strings for the keys here and in other spots? I think just regular variables would be more stable.

Reply:
It would be more stable because we would be able to catch typos more easily and we can use dot notation.

Reply:
'key' because it represents the key value of objects we get from the poverty/WIC etc data
'color' for which color to use
'value' for the actual numeric value of that feature/race

I couldnt find a better name. lol

a prop is passed into Legend component which is an array (legend) containing objects with above strings as keys, using which the Legend component is created.